src/java.base/share/classes/sun/nio/ch/OptionKey.java

```diff
@@ -28,21 +28,5 @@
 /**
  * Represents the level/name of a socket option
  */
-
-class OptionKey {
-    private int level;
-    private int name;
-
-    OptionKey(int level, int name) {
-        this.level = level;
-        this.name = name;
-    }
-
-    int level() {
-        return level;
-    }
-
-    int name() {
-        return name;
-    }
+record OptionKey(int level, int name) {
```

Comment:
This changes equality, will need to examine the usages in generated code to check that it is okay.

Reply:
`OptionKey` is used only as a value in HashMap `sun.nio.ch.SocketOptionRegistry.LazyInitialization#options`.
The only usage of this map is `sun.nio.ch.SocketOptionRegistry#findOption`
https://github.com/openjdk/jdk/blob/e8a37b90db8dca4dc3653970b2d66d2faf8ef452/src/java.base/share/classes/sun/nio/ch/SocketOptionRegistry.java.template#L144-L147
Returned `OptionKey` is then only compared with `null` and then deconstructed to `level`/`name` parts.

Hence, changing equiality doesn't change the behavior.

Reply:
> Hence, changing equiality doesn't change the behavior.

We need to double check to be 100% sure, it's a change that probably shouldn't be in this PR.

Reply:
Ok, reverted